spec/factories/stages.rb

```diff
@@ -16,10 +16,18 @@
     factory :playoff_stage do
       level { rand(10) }
       transient do
+        match_type { :running_playoff_match }
         match_count { 4 }
       end
       after(:create) do |stage, evaluator|
-        stage.matches = create_list(:match, evaluator.match_count)
+        # match_count -1 automatically generates 2 ^ stage.level matches
+        # (as this would be the amount of stages present in the real world)
+        stage.matches = create_list(evaluator.match_type,
+                                    evaluator.match_count == -1 ? 2**stage.level : evaluator.match_count)
+        stage.matches.each_with_index do |match, i|
+          match.position = i
+          match.save!
+        end
```

Comment:
you should probably `.save!` the `stage`.
(that will probably save the matches as well)

Reply:
done